RFCs/0011-DeletionPolicy.md

```diff
@@ -0,0 +1,262 @@
+            "AllowedValues": ["Prod", "Staging", "Dev"]
+        }
+    },
+    "Mappings" : {
+        "RegionMap" : {
+            "us-east-1" : { 
+                "DeletionPolicy": "Retain",
+                "UpdateReplacePolicy": "Retain"
+            },
+            "us-west-1" : { 
+                "DeletionPolicy": "Delete",
+                "UpdateReplacePolicy": "Delete"
+            },
+            "eu-west-1" : { 
+                "DeletionPolicy": "Retain",
+                "UpdateReplacePolicy": "Delete"
+            },
+            "ap-southeast-1" : { 
+                "DeletionPolicy": "Delete",
+                "UpdateReplacePolicy": "Delete"
+            }
+        }
+    },
+    "Resources": {
+        "Bucket": {
+            "Type": "AWS::S3::Bucket",
+            "DeletionPolicy": { 
+                "Fn::FindInMap" : [
+                    "RegionMap", 
+                    { "Ref" : "AWS::Region" },
+                    "DeletionPolicy"
+                ]
+            },
+            "UpdateReplacePolicy": { 
+                "Fn::FindInMap" : [
+                    "RegionMap", 
+                    { "Ref" : "AWS::Region" },
+                    "UpdateReplacePolicy"
+                ]
+            }
+        }
+    }
+}
+```
+
+# Motivation
+
+CloudFormation customers expect to use conditions and intrinsic functions as a value for a `DeletionPolicy` and `UpdateReplacePolicy` Resource attributes as they can do elsewhere in the template.
+Main ask from our customers was to allow them to vary DeletionPolicy and UpdateReplacePolicy values based on development stage, which tends to be passed in as a parameter, e.g., allow resource deletion in Dev stages but not Prod. This RFC proposes to support not only such use case but also other intrinsic function usages and Pseudo-parameter references.
+
+* References
+    * https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/162 (148 thumbs up)
+    * https://github.com/aws/aws-cli/issues/3825
+
+# Details
+* Following intrinsic functions will be supported.
+    * [Ref](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html)
+    * [Fn::FindInMap](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap.html)
+    * [Fn::If](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-conditions.html#intrinsic-function-reference-conditions-if)
+
+
+* Following Pseudo-Parameters references will be supported
+    * `AWS::AccountId`, `AWS::Region`, `AWS::Partition`
+
+# Limitation
+
+Does not support expressions containing references to *[Resource properties](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html)* or *certain pseudo parameters*
+* Unsupported Pseudo-Parameter References
+    * `AWS::StackId`, `AWS::StackName`, `AWS::NotificationArns`, `AWS::URLSuffix`, `AWS::NoValue`
```

Comment:
Guessing resource properties are excluded since their values might not be known until during stack operations, but might be insightful to elaborate on why certain psuedo parameters won't be supported since their values seem assigned by the beginning of stack operations

Reply:
We are focusing on the ones that are most important to fulfilling the use case. All others are marked as not supported to keep our design options open.